### PR TITLE
docs(upgrading): update version upgrade summary

### PR DESCRIPTION
**Pull request by suchapalaver**
Bringing `upgrading.md` up to date.

[CHRON-296](https://blockchaintp.atlassian.net/browse/CHRON-296)

Signed-off-by: Joseph Livesey [joseph.livesey@btp.works](mailto:joseph.livesey@btp.works)

[CHRON-296]: https://blockchaintp.atlassian.net/browse/CHRON-296?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docs/upgrading.md [hunk 1]
```diff
@@ -1,5 +1,118 @@
...
+`AuthId` with a verified external ID can then be found in the GraphQL context.
+
+We also converted the `JWT` `AuthId` variant to be a structure of both a json-pointer
+-derived scalar and JWT claims.
+
+### Anonymous Identity Variant
+
+By default, Chronicle allows unauthenticated access to the GraphQL API, classifying
+such a user as "anonymous." Adds an identity variant emanating from GraphQL or CLI,
+not authorized via Auth0, allowing for the option to pass a transactor key as a CLI
```
Comment (mtbc) [inline docs/upgrading.md:19]: No need to mention Auth0 specifically, deployments may be using other providers.

File: docs/upgrading.md [hunk 2]
```diff
@@ -1,5 +1,118 @@
...
+TransactionHandlers during tests and in `devmode`.
+
+## `prov:wasAttributedTo`
+
+Chronicle now supports the `wasAttributedTo` PROV-O relationship.
+
+## Endpoint offering JSON by IRI
+
+Now, running Chronicle with the `serve-api` command, users can visit
+<http://localhost:9982/data/chronicle:agent:my-agent-id> and see a JSON summary.
```
Comment (mtbc) [inline docs/upgrading.md:83]: also for entity and activity

File: docs/upgrading.md [hunk 3]
```diff
@@ -1,5 +1,118 @@
...
+### `AuthId` Identity Type
+
+We introduced an identity union in this update, which offers a versatile and flexible
+approach to identity management. This new feature can be created from any of the
+following options:
+
+- Chronicle root key, identified by the agent name "chronicle"
+- An agent along with its corresponding key pair
+- A pointer that refers to the external id of an agent and its key pair through
+  dereferencing a JWT structure
```
Comment (mtbc) [inline docs/upgrading.md:111]:
```suggestion
- an agent along with its corresponding key pair
- a pointer into a JWT structure, for retrieving the external id to use for the agent and its key pair.
```

File: docs/upgrading.md [hunk 4]
```diff
@@ -1,5 +1,118 @@
...
+The `OpaExecutor` interface takes a Chronicle `AuthId` identity and an arbitrary
+JSON context, which it evaluates against an OPA policy. The `WasmtimeOpaExecutor`
+evaluates OPA policies compiled to .wasm for execution.
+
+The `PolicyLoader` interface offers two implementations. The `SawtoothPolicyLoader`
+implementation loads an OPA policy rule that is compiled to .wasm from a Sawtooth
+address. The `CliPolicyLoader` implementation provides support for loading an OPA
+policy .wasm file via CLI arguments for a file path and policy entrypoint.
+
+Chronicle integrates OPA execution at its GraphQL API request point and in its Transaction
```
Comment (ryanbtp) [inline docs/upgrading.md:41]: endpoint

File: docs/upgrading.md [hunk 5]
```diff
@@ -1,5 +1,118 @@
...
+with the Chronicle TP. Tested at the protobuf level, this feature is designed to
+operate independently of the rest of the Chronicle project.
+
+## To 0.5x
+
+### Move from SQLite to PostgreSQL
+
+Chronicle now utilizes PostgreSQL instead of SQLite as its local database.
+This change brings several benefits, such as increased scalability and better
+performance for large-scale data management.
```
Comment (ryanbtp) [inline docs/upgrading.md:100]: As an aside this is noticeable with the ETL on the discover project that's using a fake ledger. Chronicle is blessed performance wise as it is almost certainly dependent on something a lot slower than it so is always going to look good.

**Review by ryanbtp — CHANGES_REQUESTED** [review]

**Review by ryanbtp — APPROVED** [review]

**Pull request merged**